### PR TITLE
Implement browser history API based navigation, preventing page reloads between routings

### PR DESCRIPTION
**Pull request by jacob-js**
https://github.com/cypht-org/cypht/issues/1144

Checklist:
- [x] The sidebar item's active state updates when navigating
- [x] Ensure handlers for refresh links are applied so the browser doesn’t reload on click
- [x] Sorting and filtering work
- [x] Fix combined folder content not being loaded upon navigation
- [x] Can display flagged contents
- [x] Cronjob to reload active folder's content in the background
- [x] Refresh the list during navigation; at least in the background.
- [x] FIX: Navigating to a route, the data from a previous route gets fetched.
- [x] Fix unit tests relative to the js files output
- [x] ~~When clicking on a message link while contents are still being fetched in the background, the requests should be aborted so the navigation is performed faster~~ (Aborting the requests on the client doesn't help because, once the requests are sent, the server processes them synchronously, blocking other eventual requests. A solution could be processing the requests server-side asynchronously.)
- [x] Can requests be processed asynchronously? Let's keep track of this here: https://github.com/cypht-org/cypht/issues/1279

### CODE REVIEW

**Comment by marclaporte** [comment]
@jacob-js A test is failing.

**Comment by jacob-js** [comment]
> @jacob-js A test is failing.

Locally the same test is succeeding, not sure why in the CI it is failing.  @kroky could you have some insights?
Here is what I have when I execute that test:

```shell
LOGIN TESTS
 - loading site
 - logging in
 - finding element by name username
 - finding element by name password
 - finding element by selector input[value=Login]
 - waiting for page by tag name: body ...
 - finding element by class sys_messages
bad_login_values PASSED
 - loading site
 - logging in
 - finding element by name username
 - finding element by name password
 - finding element by selector input[value=Login]
 - waiting for page by tag name: body ...
 - finding element by class login_form
missing_password PASSED
 - loading site
 - logging in
 - finding element by name username
 - finding element by name password
 - finding element by selector input[value=Login]
 - waiting for page by tag name: body ...
 - finding element by class login_form
missing_username PASSED
 - loading site
 - logging in
 - finding element by name username
 - finding element by name password
 - finding element by selector input[value=Login]
 - waiting for page by tag name: body ...
 - finding element by class login_form
missing_username_and_password PASSED
 - loading site
 - finding element by name hm_page_key
 - logging in
 - finding element by name username
 - finding element by name password
 - finding element by selector input[value=Login]
 - waiting for page by tag name: body ...
 - finding element by class login_form
bad_login_key PASSED
 - loading site
 - logging in
 - finding element by name username
 - finding element by name password
 - finding element by selector input[value=Login]
 - waiting for page by class name: main_menu ...
 - finding element by class content_title
good_login PASSED
 - logging out
 - waiting for page by tag name: body ...
 - finding element by class sys_messages
good_logout PASSED

7 of 7 PASSED
```

**Review by kroky — APPROVED** [review]
Changes look good. I am also puzzled by the failing CI test - it is either an unsaved change that is triggering the password dialog on logout or somehow sys_messages become inaccessible for selenium. I think that needs to be debugged more in the pipeline itself, since it can't be reproduced locally. I'd try swapping logout with logout_no_save first.

**Comment by jacob-js** [comment]
> Changes look good. I am also puzzled by the failing CI test - it is either an unsaved change that is triggering the password dialog on logout or somehow sys_messages become inaccessible for selenium. I think that needs to be debugged more in the pipeline itself, since it can't be reproduced locally. I'd try swapping logout with logout_no_save first.

The incoherence in the test results was due to different modes used in the test environment. The CI doesn't run in debug mode, whereas locally, the app is executed in debug mode.

**Comment by jacob-js** [comment]
@kroky could you look at the changes one more time please?

File: modules/core/output_modules.php [hunk 1]
```diff
@@ -554,20 +554,36 @@ protected function output() {
...
+
+                    if (is_readable(sprintf("%ssite.js", $name))) {
+                        $res .= '<script type="text/javascript" src="'.WEB_ROOT.sprintf("%ssite.js", $rel_name).'"></script>';
+                    }
                 }
             }
             if ($core) {
-                $res = '<script type="text/javascript" src="'.WEB_ROOT.sprintf("%ssite.js", $core).'"></script>'.$res;
+                $res = // Load navigation utilities used by subsequent modules' handlers
+                '<script type="text/javascript" src="modules/core/navigation/utils.js"></script>'.
```
Comment (kroky) [inline modules/core/output_modules.php:576]: Isn't WEB_ROOT preferred here as well?

**Pull request merged**